src/main/java/fr/uge/lp/databackbonepoc/linkresearch/comparator/DBComparator.java

```diff
@@ -41,20 +41,23 @@ public DBComparator(
    * @param matcherConfig a MatcherConfiguration entity object containing the matching algorithm to use and its parameters
    * @return the DBComparator instance that can be used for comparing two values with the specified matching algorithm
    */
-  // FIXME
   public static DBComparator<?> fromMatcherConfiguration(MatcherConfiguration matcherConfig) {
     Objects.requireNonNull(matcherConfig);
-    var matchingAlgorithm = matcherConfig.getMatchingAlgorithm();
-    var algorithm = switch (matchingAlgorithm) {
-      case "strict_equality" -> ComparatorAlgorithm.STRICT_EQUALS; // handle type depending on the attributes' type
-      default -> throw new UnsupportedOperationException("Unsupported matching algorithm");
+    var matchingAlgorithm = matcherConfig.getMatchingAlgorithm().getName().algorithm();
+    // FIXME handle type : it depends on the algorithm's type
+    return switch(matchingAlgorithm) {

```

Comment:
```suggestion
    return switch (matchingAlgorithm) {
```

Si on veut être tatillon